db/compaction/compaction.cc

```diff
@@ -577,6 +619,8 @@ std::unique_ptr<SstPartitioner> Compaction::CreateSstPartitioner() const {
   context.output_level = output_level_;
   context.smallest_user_key = smallest_user_key_;
   context.largest_user_key = largest_user_key_;
+  context.output_next_level_segments =
+      CreateSegmentsForLevel(output_level_ + 1);
```

Comment:
Should consider start and end key.

Reply:
The start key and end key of this compaction can be accessed during creating segments, could you describe more details about the advice?

Reply:
I see.